accord-core/src/main/java/accord/local/CommandStores.java

```diff
@@ -351,6 +350,7 @@ private synchronized TopologyUpdate updateTopology(Node node, Snapshot prev, Top
             Ranges removeRanges = subtracted.slice(current, Minimal);
             if (!removeRanges.isEmpty())
             {
+                // TODO (required): This is updating the a non-volatile field in the previous Snapshot, why modify it at all, even with volatile the guaranteed visibility is weak even with mutual exclusion
```

Comment:
@belliottsmith Why does this update the previous snapshot in place? We can't guarantee anything in flight reads it so they might as well read it out of the current snapshot?

Can we make a copy to avoid the mutability issue?

Reply:
I'm not sure what you mean? This is the snapshot we're updating, inside the snapshot_holder. This is safe published because all of the fields are final.